types/scene/wlr_scene.c

```diff
@@ -1231,6 +1278,41 @@ static void scene_entry_render(struct render_list_entry *entry, const struct ren
 		};
 		fx_render_pass_add_rect(data->render_pass, &rect_options);
 		break;
+	case WLR_SCENE_NODE_SHADOW:;
+		struct wlr_scene_shadow *scene_shadow = wlr_scene_shadow_from_node(node);
+
+		struct wlr_box window_box;
+		int window_x = 0, window_y = 0, window_width = 0, window_height = 0, window_corner_radius = 0;
+		struct wlr_scene_node *parent_node = wl_container_of(node->link.prev, parent_node, link);
+		if (parent_node->type == WLR_SCENE_NODE_BUFFER || parent_node->type == WLR_SCENE_NODE_RECT) {
```

Comment:
This should preferably be done in a loop due to get all above siblings :)

Reply:
Would we want to do this? If a sibling is transparent we should see the main surface's shadow beneath it

Reply:
Will do as follow-up, for now will rely on the next node